students/drovdahl/lesson03/strformat_lab.py

```diff
@@ -0,0 +1,102 @@
+#! /usr/local/bin/python3
+
+
+'''
+Perform a number of string formatting exercises
+'''
+import os
+
+
+def task_1():
+    '''
+    produce 'file_002 :   123.46, 1.00e+04, 1.23e+04'
+    '''
+    seq = (2, 123.4567, 10000, 12345.67)
+    fname = seq[0]
+    number1 = seq[1]
+    number2 = seq[2]
+    number3 = seq[3]
+    print(f'file_{fname:0>3d}:  {number1:.2f}, {number2:.2e}, {number3:.2e}')
+
+
+def task_2():
+    '''
+    produce the same results as task_1 using an alternate type of format string
+    '''
+    seq = (2, 123.4567, 10000, 12345.67)
+    print('\n\n')
+    print('file_{:0>3}:  {:.2f}, {:.2e}, {:.2e}'.format(seq[0], seq[1], seq[2],
+          seq[3]))
+    # here's a shortened version
+    print('file_{:0>3}:  {:.2f}, {:.2e}, {:.2e}'.format(*seq))
+
+
+def task_3(seq):
+    '''
+    rewrite "the 3 numbers are: {:d}, {:d}, {:d}".format(1,2,3)
+    to take an arbitrary number of values.
+    '''
+    length = len(seq)
+    values_list = ['{}'] * length
+    values = ', '.join(values_list)
+    print('\n\n')
+    print(('the {} values are ' + values).format(length, *seq))
+
+
+def task_4(seq):
+    '''
+    given tuple( 4, 30, 2017, 2, 27)
+    use string formating to print:
+    '02 27 2017 04 30'
+    '''
+    print('\n\n')
+    print(f'{seq[3]:0>2d} {seq[4]} {seq[2]} {seq[0]:0>2d} {seq[1]}')
```

Comment:
What if day (e.g., 30 here) or minute (e.g., 27 here) is a single digit?

Reply:
I added appropriate formatting and tested these scenarios.  Updates will show up in the next PR